### DISCUSSION TITLE
model diagnostics - gradients > 0.001

### DISCUSSION THREAD

**Discussion by lysoifer** [sdmTMB/sdmTMB#358] [Q&A] [answered]
Hi, 
I am working on using sdmTMB to model patterns of vertebrate traits. For some of my models, the sanity() check reveals gradients greater than 0.001 but states that the non-linear minimizer successfully converged and the hessian matrix is positive definite. I have tried running extra optimizations on the model, but the gradients remain greater than 0.001. Is this cause for concern or does the successful convergence and positive definite hessian matrix indicate the model is okay?

Thanks,
Lydia

**Comment by Lewis-Barnett-NOAA** [comment, marked as answer]
Hi Lydia,

This is a common question, thanks for bringing it up.  The 0.001 is fairly
arbitrary, so I would say that you are probably good to go if you've run a
few extra optimization loops already and all the other diagnostics check
out, along with the parameter estimates and their variance being
reasonable.  If the maximum final gradient is still much > 0.001 then I
would consider simplifying the model structure.


On Fri, Jul 19, 2024 at 4:17 AM Lydia Soifer ***@***.***>
wrote:

> Hi,
> I am working on using sdmTMB to model patterns of vertebrate traits. For
> some of my models, the sanity() check reveals gradients greater than 0.001
> but states that the non-linear minimizer successfully converged and the
> hessian matrix is positive definite. I have tried running extra
> optimizations on the model, but the gradients remain greater than 0.001. Is
> this cause for concern or does the successful convergence and positive
> definite hessian matrix indicate the model is okay?
>
> Thanks,
> Lydia
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pbs-assess/sdmTMB/discussions/358>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKMJP44FL2CBQGDXVTWFV3ZNDYTNAVCNFSM6AAAAABLEMT2JKVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWHE2TEOJSGA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Lewis Barnett, PhD (he/him)
Research Fish Biologist

NOAA Fisheries, Alaska Fisheries Science Center
7600 Sand Point Way NE, Bldg 4
Seattle, Washington 98115
Google Voice: (206) 526-4111
